### PR TITLE
chore: set the plugin visibility

### PR DESCRIPTION
**Pull request by FeiWang1119**
set the plugin in tray visibility

Log:

### CODE REVIEW

File: panels/dock/dockiteminfo.cpp [hunk 1]
```diff
@@ -10,22 +10,25 @@ QDebug operator<<(QDebug argument, const DockItemInfo &info)
...
-             << "icon_light:" << info.iconLight << "icon_dark:" << info.iconDark << "visible:" << info.visible;
+             << "dcc_icon:" << info.dccIcon << "visible:" << info.visible;
     return argument;
 }
 
 QDBusArgument &operator<<(QDBusArgument &arg, const DockItemInfo &info)
 {
     arg.beginStructure();
-    arg << info.name << info.displayName << info.itemKey << info.settingKey << info.iconLight << info.iconDark << info.visible;
+    qWarning() << info;
```
Comment (tsic404) [inline panels/dock/dockiteminfo.cpp:20]:
```suggestion
```

File: panels/dock/pluginmanagerextension.cpp [hunk 2]
```diff
@@ -15,7 +15,7 @@
 #include <QJsonObject>
 #include <QJsonParseError>
 
-PluginSurface::PluginSurface(PluginManager* manager, const QString& pluginId, const QString& itemKey, const QString &displayName, int pluginFlags, int pluginType, int sizePolicy, QWaylandSurface *surface, const QWaylandResource &resource)
+PluginSurface::PluginSurface(PluginManager* manager, const QString& pluginId, const QString& itemKey, const QString &displayName, int pluginFlags, int pluginType,  int sizePolicy, QWaylandSurface *surface, const QWaylandResource &resource)
```
Comment (tsic404) [inline panels/dock/pluginmanagerextension.cpp:18]:
```suggestion
PluginSurface::PluginSurface(PluginManager* manager, const QString& pluginId, const QString& itemKey, const QString &displayName, int pluginFlags, int pluginType, int sizePolicy, QWaylandSurface *surface, const QWaylandResource &resource)
```

File: panels/dock/pluginmanagerextension.cpp [hunk 3]
```diff
@@ -103,6 +108,12 @@ void PluginSurface::plugin_mouse_event(QtWaylandServer::plugin::Resource *resour
     Q_EMIT recvMouseEvent((QEvent::Type)type);
 }
 
+void PluginSurface::plugin_dcc_icon(Resource *resource, const QString &type)
```
Comment (tsic404) [inline panels/dock/pluginmanagerextension.cpp:111]:
```suggestion
void PluginSurface::plugin_dcc_icon(Resource *resource, const QString &icon)
```

File: panels/dock/pluginmanagerextension.cpp [hunk 4]
```diff
@@ -103,6 +108,12 @@ void PluginSurface::plugin_mouse_event(QtWaylandServer::plugin::Resource *resour
...
+{
+    qInfo() << "dcc_icon:" << type;
+    m_dccIcon = type;
```
Comment (tsic404) [inline panels/dock/pluginmanagerextension.cpp:114]:
```suggestion
    m_dccIcon = icon;
```

File: panels/dock/pluginmanagerextension_p.h [hunk 5]
```diff
@@ -100,7 +102,8 @@ class PluginSurface : public QWaylandShellSurfaceTemplate<PluginSurface>, public
     void recvMouseEvent(QEvent::Type type);
 
 protected:
-    virtual void plugin_mouse_event(Resource *resource, int32_t type) override;
+    virtual void plugin_mouse_event(Resource *resource, int32_t type) override;;
+    virtual void plugin_dcc_icon(Resource *resource, const QString &type) override;
```
Comment (tsic404) [inline panels/dock/pluginmanagerextension_p.h:106]:
```suggestion
    virtual void plugin_dcc_icon(Resource *resource, const QString &icon) override;
```

File: panels/dock/pluginmanagerextension_p.h [hunk 6]
```diff
@@ -100,7 +102,8 @@ class PluginSurface : public QWaylandShellSurfaceTemplate<PluginSurface>, public
     void recvMouseEvent(QEvent::Type type);
 
 protected:
-    virtual void plugin_mouse_event(Resource *resource, int32_t type) override;
+    virtual void plugin_mouse_event(Resource *resource, int32_t type) override;;
```
Comment (tsic404) [inline panels/dock/pluginmanagerextension_p.h:105]:
```suggestion
    virtual void plugin_mouse_event(Resource *resource, int32_t type) override;
```

File: panels/dock/dockiteminfo.cpp (hunk 1 above)
Comment (FeiWang1119) [reply panels/dock/dockiteminfo.cpp:20]: done

File: panels/dock/pluginmanagerextension.cpp (hunk 2 above)
Comment (FeiWang1119) [reply panels/dock/pluginmanagerextension.cpp:18]: done

File: panels/dock/pluginmanagerextension.cpp (hunk 3 above)
Comment (FeiWang1119) [reply panels/dock/pluginmanagerextension.cpp:111]: done

File: panels/dock/pluginmanagerextension.cpp (hunk 4 above)
Comment (FeiWang1119) [reply panels/dock/pluginmanagerextension.cpp:114]: done

File: panels/dock/pluginmanagerextension_p.h (hunk 5 above)
Comment (FeiWang1119) [reply panels/dock/pluginmanagerextension_p.h:106]: done

File: panels/dock/pluginmanagerextension_p.h (hunk 6 above)
Comment (FeiWang1119) [reply panels/dock/pluginmanagerextension_p.h:105]: done

**Review by tsic404 — APPROVED** [review]

**Pull request merged**